ddtrace/context.py

```diff
@@ -0,0 +1,85 @@
+import logging
+import threading
+
+
+log = logging.getLogger(__name__)
+
+
+class Context(object):
+    """
+    Context is used to keep track of a hierarchy of spans for the current
+    execution flow.
+
+    TODO: asyncio is not thread-safe by default. The fact that this class is
+    thread-safe is an implementation detail. Avoid mutex usage when the Context
```

Comment:
Usually they should be avoided in async stuff because it's supposed to be single threaded. Actually I'm keeping it to have one single ``Context`` so yes, probably we can just keep it.